src/main/java/internetshop/dao/hibernate/BucketDaoHibernateImpl.java

```diff
@@ -0,0 +1,66 @@
+package internetshop.dao.hibernate;
+
+import internetshop.annotation.Dao;
+import internetshop.dao.BucketDao;
+import internetshop.model.Bucket;
+import internetshop.util.HibernateUtil;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class BucketDaoHibernateImpl implements BucketDao {
+    @Override
+    public Bucket create(Bucket bucket) {
+        Transaction transaction = null;
+        Long bucketId = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
try to use block finally and close session here in methods create, update, delete